File: yt-message-admin/yt-message-admin-server/src/main/java/com/yt/message/admin/server/pojo/entity/Template.java
package com.yt.message.admin.server.pojo.entity;

import com.baomidou.mybatisplus.annotation.TableId;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.io.Serializable;

/**
 * <p>
 * 消息模板
 * </p>
 *
 * @author yt
 * @since 2024-11-26
 */
@EqualsAndHashCode(callSuper = true)
@Data
public class Template extends  BaseEntity implements Serializable  {

    private static final long serialVersionUID = 1L;

    /**
     * 模板id
     */
    @TableId("template_id")
    private Long templateId;

    /**
     * 模板名
     */
    private String templateName;

    /**
     * 平台id
     */
    private Long platformId;

    /**
     * 对应第三方平台模板id
     */
    private String platformTemplateId;

    /**
     * message_usage
     */
    private Integer messageUsage;

    /**
     * 是否需要持久化
     */
    private Integer requirePersist;

    /**
     * 是否需要验证重复消费
     */
    private Integer requireRepeatConsume;


    /**
     * 是否需要限流
     */

    private Integer requireRateLimit;
    /**
     * 平台服务限流策略
     */
    private Integer rateLimitStrategy;

    /**
     * 发送账户
     */

    private String sendAccount;

    /**
     * 用户名
     */

    private String username;

    /**
     * 密码
     */
    private String password;

    /**
     * 是否需要ssl发送
     */

    private Integer requireSsl;


    /**
     * 标题(可包含占位符{})
     */

    private String subject;


    /**
     * 正文(可包含占位符{})
     */

    private String content;

    /**
     * 正文内容是否是html格式
     */

    private Integer requireHtml;


    private Integer version;



}

File: yt-message-admin-ui/vue-vben-admin/apps/web-ele/src/api/system/template.ts
import type { BasePageReqParams, PageRsp } from "../common/types";
import { requestClient } from '#/api/request';

export namespace templateApi {
  export const baseUri = '/template'

  /** 分页参数 */
  export interface pageParams extends BasePageReqParams {

  }
  export interface TemplateRsp {
    /**
  * 模板id
  */

    templateId: string;

    /**
     * 模板名
     */
    templateName: string

    /**
     * 平台id
     */
    platformId: string

    /**
     * 对应第三方平台模板id
     */
    platformTemplateId: string

    /**
     * message_usage
     */
    messageUsage: number

    /**
     * 是否需要持久化
     */
    requirePersist: number

    /**
     * 是否需要验证重复消费
     */
    requireRepeatConsume: number

  

    /**
     * 是否需要限流
     */

    requireRateLimit: number

    rateLimitStrategy?: number;

    /**
     * 发送账户
     */

    sendAccount: string

    /**
     * 用户名
     */
    username: string

    /**
     * 密码
     */
    password: string

    /**
     * 是否需要ssl发送
     */

    requireSsl: number


    /**
     * 标题(可包含占位符{})
     */

    subject: string


    /**
     * 正文(可包含占位符{})
     */

    content: string

    /**
     * 正文内容是否是html格式
     */

    requireHtml: number

    messageType?: number
    version:number;
  }
}

export async function page(data: templateApi.pageParams) {
  return requestClient.post<PageRsp<templateApi.TemplateRsp>>(templateApi.baseUri + '/page', data);
}
export async function update(data: any) {
  return requestClient.put(templateApi.baseUri, data)
}
export async function create(data: any) {
  return requestClient.post(templateApi.baseUri, data)
}
export async function remove(id : string) {
  return requestClient.delete(templateApi.baseUri + '/' + id)
}
export async function sendMessage(data: any) {
  return requestClient.post(templateApi.baseUri+'/send', data)
}
